src/tests/ftest/util/data_mover_test_base.py

```diff
@@ -1195,81 +1195,91 @@ def run_datamover(self, test_desc=None,
 
         return result
 
-    def run_dm_activities_with_ior(self, tool, create_dataset=False):
+    def run_dm_activities_with_ior(self, tool, create_dataset=False, pool=None):
         """Generic method to perform varios datamover activities
            using ior
         Args:
             tool(str): specify the tool name to be used
             create_dataset(bool): boolean to create initial set of
                                   data using ior. Defaults to False.
+            pool(TestPool): Pool object.Defaults to None
         """
         # Set the tool to use
         self.set_tool(tool)
 
+
         if create_dataset:
             # create initial datasets
-            pool1 = self.create_pool()
-            cont1 = self.create_cont(pool1, oclass=self.ior_cmd.dfs_oclass.value)
+            if not pool:
+                pool = self.create_pool()
+            cont1 = self.create_cont(pool, oclass=self.ior_cmd.dfs_oclass.value)
 
             # update and run ior on cont1
             self.run_ior_with_params(
                 "DAOS", self.ior_cmd.test_file.value,
-                pool1, cont1)
+                pool, cont1)
+        else:
+            if not pool:
+                pool = self.pool[0]
+            cont1 = self.container[-1]
```

Comment:
I think this could be tricky under certain circumstances. E.g. `self.container[-1]` might not even be in `self.pool[0]`. It's also disjoint that we're using the first pool, but the last container. Furthermore, `pool` is an optional argument to this function but `cont` is not. This works for these particular tests, but would be messy if this function needed to be used in a different scenario. So far it looks like the test case always passes create_dataset=True, so it is even necessary to support create_dataset=False at this time?

Reply:
I have added 'cont' as an optional argument as well, so I guess that should take care of the concerned mentioned above. But I would like to point that if we are using any of the datamover features and using pre-existing pool, I would say it should consist of at least one container (i.e self.container[-1]). But I have still added an option 'cont' though.
And regarding 'create_dataset=False', if it's fine I would like to keep it as is. The changes which will be using it is not far. It should be pushed in a day or 2.

Reply:
I think the `cont` argument is good. My main concern was that this reminded my of the IOR test base, where there are several functions that work in a very specific way that isn't configurable (IOR test base is better now at least). Thanks!